Sources/AWSXRayInstrument/Instrument.swift

```diff
@@ -0,0 +1,40 @@
+import AWSXRayRecorder
+import Baggage
+import Dispatch // DispatchTime
+import Instrumentation
+import NIOHTTP1 // HTTPHeaders
+import NIOInstrumentation // HTTPHeadersExtractor, HTTPHeadersInjector
+
+extension XRayRecorder: TracingInstrument {
+    public func extract<Carrier, Extractor>(_ carrier: Carrier, into baggage: inout BaggageContext, using extractor: Extractor) where Carrier == Extractor.Carrier, Extractor: ExtractorProtocol {
+        guard
+            let headers = carrier as? HTTPHeaders,
+            let extractor = extractor as? HTTPHeadersExtractor,
+            let tracingHeader = extractor.extract(key: AmazonHeaders.traceId, from: headers)
+        else {
+            return
+        }
+
+        if let context = try? XRayContext(tracingHeader: tracingHeader) {
+            baggage.xRayContext = context
+        }
+    }
+
+    public func inject<Carrier, Injector>(_ baggage: BaggageContext, into carrier: inout Carrier, using injector: Injector) where Carrier == Injector.Carrier, Injector: InjectorProtocol {
+        guard
+            let context = baggage.xRayContext,
+            var headers = carrier as? HTTPHeaders, // TODO: ! this makes a copy
```

Comment:
this breaks one of the tests, will revisit in the evening

Reply:
Same thing for `inject`. Why the cast to `HTTPHeaders`?

Reply:
fixed, thanks!

Reply:
Nice, glad to see this part of the API turned out useful 😊